src/servicebus/Elsa.AzureServiceBus/Activities/ServiceBusSignaled.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using Elsa.Extensions;
+using Elsa.Persistence;
+using Newtonsoft.Json.Linq;
+using Elsa.AzureServiceBus.Models;
+using Elsa.AzureServiceBus.Services;
+
+namespace Elsa.AzureServiceBus.Activities
+{
+    [ActivityDefinition(
+        Category = "Azure",
+        Description = "Send a message and halt workflow execution until the specified signal is received.",
+         RuntimeDescription = "x => !!x.state.queue ? `<strong>${x.state.dataExpression.expression}</strong> -> <strong>${x.state.queue}</strong><br/>${x.state.dataExpression.syntax}` : x.definition.description",
+        Icon = "fas fa-traffic-light",
+        Outcomes = "x => !!x.state.signals ? x.state.signals : []")]
+
+    public class ServiceBusSignaled : Activity
+    {
+        public const string INPUT_VARIABLE_NAME = "AzureServiceBus_Signal";
+
+        private readonly IWorkflowExpressionEvaluator _expressionEvaluator;
+        private readonly IServiceBusClientFactory _serviceBusProvider;
+        private readonly ITokenService _tokenService;
+
+        public ServiceBusSignaled(IWorkflowExpressionEvaluator expressionEvaluator, IServiceBusClientFactory serviceBusProvider, ITokenService tokenService)
+        {
+
+            _serviceBusProvider = serviceBusProvider;
+            _expressionEvaluator = expressionEvaluator;
+            _tokenService = tokenService;
+        }
+
+        [ActivityProperty(
+
+            Hint = "The queue to send to"
+        )]
+        public string Queue
+        {
+            get => GetState(() => string.Empty);
+            set => SetState(value);
+        }
+
+        [ActivityProperty(Hint = "Message data")]
+        public WorkflowExpression<object> DataExpression
+        {
+            get => GetState(() => new WorkflowExpression<object>(LiteralEvaluator.SyntaxName, ""));
+            set => SetState(value);
+        }
+
+        [ActivityProperty(
+            Type = ActivityPropertyTypes.List,
+            Hint = "An expression that evaluates to the name of the signal to wait for."
+        )]
+        public HashSet<string> Signals
+        {
+            get => GetState(() => new HashSet<string> { "Approved" });
+            set => SetState(value);
+        }
+
+        [ActivityProperty(Hint = "The name of the queue to receive the message on")]
+        public string ConsumerName
+        {
+            get => GetState(() => "*");
+            set => SetState(value);
+        }
+
+
+
+        protected override async Task<ActivityExecutionResult> OnExecuteAsync(WorkflowExecutionContext context, CancellationToken cancellationToken)
+        {
+            var data = await context.EvaluateAsync(DataExpression, cancellationToken);
+
+            if (string.IsNullOrWhiteSpace(Queue) == false)
+            {
+                //we need to generate a signal for each of the signals
```

Comment:
When writing code comments, please use properly formatted English sentences that start with a capital letter and end with a full stop. Also add a space between the first two slashes and the start of the sentence. E.g. `// We need to generate a signal for each of the signals.`

Please apply everywhere in this PR. Thank you.